frame/democracy/src/lib.rs

```diff
@@ -1578,6 +1682,52 @@ impl<T: Config> Pallet<T> {
 		// `Compact<u32>`.
 		decode_compact_u32_at(&<DepositOf<T>>::hashed_key_for(proposal))
 	}
+
+	/// Return a proposal of an index.
+	fn proposal(index: PropIndex) -> Result<(PropIndex, BoundedCallOf<T>, T::AccountId), Error<T>> {
+		PublicProps::<T>::get()
+			.into_iter()
+			.find(|(prop_index, _, _)| prop_index == &index)
+			.ok_or(Error::<T>::ProposalMissing)
+	}
+
+	/// Clear metadata if exist for a given owner.
+	fn clear_metadata(owner: MetadataOwner) {
+		if let Some(hash) = MetadataOf::<T>::take(&owner) {
+			Self::deposit_event(Event::<T>::MetadataCleared { owner, hash });
+		}
+	}
+
+	/// Reset the metadata of an `owner` to a `new_owner`.
+	fn reset_metadata(owner: MetadataOwner, new_owner: MetadataOwner) {
```

Comment:
Shall we change the method's name as well for consistency?
`fn transfer_metadata()`

Reply:
yes, I will rename

Reply:
updated